From 01c5d06be3e3d5e15eb80599758c0df7a6331854 Mon Sep 17 00:00:00 2001
From: Sailaja <sbandarupalli@habana.ai>
Date: Wed, 04 Aug 2021 14:56:17 +0300
Subject: [PATCH] [SW-50869] arch: rename greco/gaudi2.h to hl.h

this is preparation patch for defining platform specific enums in common
header file and include it in required library code

ex: PSOC_GLOBAL_CONF_RST_SRC bit definitions are changed in Gaudi2 and cant
keep enum hl_rst_src in common header file preboot.h. In up-coming patches
this enum will be moved to hl.h of each platform.

if each platform has different file names like greco.h and gaudi2.h they
cant be included in library code. So keeping same file name as hl.h

arch/gaudi2/arm/common.h -> contains common definitions across ASIC/ONIX/
PLDM targets of leagacy boot flow
arch/gaudi2/common/hl.h -> contains common definitions across arm/arc flows

Signed-off-by: Sailaja <sbandarupalli@habana.ai>
Change-Id: I9c4cb45981650b42b8501b2b0810cf96343cb295
---

diff --git a/arch/gaudi2/arc/common.c b/arch/gaudi2/arc/common.c
index 437d923..8c26d2e 100644
--- a/arch/gaudi2/arc/common.c
+++ b/arch/gaudi2/arc/common.c
@@ -36,7 +36,7 @@
 #include "preboot/thermal.h"
 #include <arcboot_export.h>
 #include <pci.h>
-#include <gaudi2.h>
+#include <hl.h>
 
 #define PRODUCT_ID_STR_SIZE	0x8
 
diff --git a/arch/gaudi2/arm/common.c b/arch/gaudi2/arm/common.c
index cc06e1c..3a7ae8c 100644
--- a/arch/gaudi2/arm/common.c
+++ b/arch/gaudi2/arm/common.c
@@ -36,7 +36,7 @@
 #include "preboot/thermal.h"
 
 #include "common.h"
-#include <gaudi2.h>
+#include <hl.h>
 
 
 #define ARC0_ICCM_BASE_ADDR		0x1000007FF7000000ull
diff --git a/arch/gaudi2/common/common.c b/arch/gaudi2/common/common.c
index de23cb1..f63e1c0 100644
--- a/arch/gaudi2/common/common.c
+++ b/arch/gaudi2/common/common.c
@@ -35,7 +35,7 @@
 #include <dcore1_xbar_dma_pll_asif_slv_regs.h>
 #include <dcore2_xbar_dma_pll_asif_slv_regs.h>
 #include <dcore3_xbar_dma_pll_asif_slv_regs.h>
-#include "gaudi2.h"
+#include "hl.h"
 
 #include <common/pci_ids.h>
 
diff --git a/arch/gaudi2/common/gaudi2.h b/arch/gaudi2/common/gaudi2.h
deleted file mode 100644
index 282941a..0000000
--- a/arch/gaudi2/common/gaudi2.h
+++ /dev/null
@@ -1,7 +0,0 @@
-/* SPDX-License-Identifier:     GPL-2.0+
- *
- * Copyright (C) 2021 HabanaLabs Ltd.
- */
-
-void set_pci_msix_pba_offset_val(void);
-void set_slv_cfg_reg_val(void);
diff --git a/arch/gaudi2/common/hl.h b/arch/gaudi2/common/hl.h
new file mode 100644
index 0000000..e46ee24
--- /dev/null
+++ b/arch/gaudi2/common/hl.h
@@ -0,0 +1,12 @@
+/* SPDX-License-Identifier:     GPL-2.0+
+ *
+ * Copyright (C) 2021 HabanaLabs Ltd.
+ */
+
+#ifndef ARCH_GAUDI2_COMMON_HL_H_
+#define ARCH_GAUDI2_COMMON_HL_H_
+
+void set_pci_msix_pba_offset_val(void);
+void set_slv_cfg_reg_val(void);
+
+#endif /* ARCH_GAUDI2_COMMON_HL_H_ */
diff --git a/arch/greco/arc/common.c b/arch/greco/arc/common.c
index 2d9cfcb..1c108f7 100644
--- a/arch/greco/arc/common.c
+++ b/arch/greco/arc/common.c
@@ -41,7 +41,7 @@
 
 #include <dcon0_tstd_masks.h>
 #include <dcon0_tstd_regs.h>
-#include <greco.h>
+#include <hl.h>
 
 
 #define HOST_GPIO_I2C_DEBUG	44
diff --git a/arch/greco/arc_elf32/common.c b/arch/greco/arc_elf32/common.c
index f89a631..e294325 100644
--- a/arch/greco/arc_elf32/common.c
+++ b/arch/greco/arc_elf32/common.c
@@ -47,7 +47,8 @@
 #include <boot_if.h>
 #include <address_space.h>
 
-#include <greco.h>
+#include "common.h"
+#include <hl.h>
 
 #define PRODUCT_ID_STR_SIZE	0x8
 
diff --git a/arch/greco/arm/common.c b/arch/greco/arm/common.c
index 14017f6..6fe5261 100644
--- a/arch/greco/arm/common.c
+++ b/arch/greco/arm/common.c
@@ -39,7 +39,7 @@
 #include <pci.h>
 
 #include "common.h"
-#include <greco.h>
+#include <hl.h>
 
 
 u32 preboot_device_id_get(void)
diff --git a/arch/greco/common/common.c b/arch/greco/common/common.c
index 63a98bc..cf713fe 100644
--- a/arch/greco/common/common.c
+++ b/arch/greco/common/common.c
@@ -37,7 +37,7 @@
 #include <pci.h>
 #include <preboot/thermal.h>
 #include <efuse.h>
-#include <greco.h>
+#include <hl.h>
 
 
 #define mmPCIE_WRAP_VUART_TX_DATA       0xC15F60
diff --git a/arch/greco/common/greco.h b/arch/greco/common/greco.h
deleted file mode 100644
index 9382b7e..0000000
--- a/arch/greco/common/greco.h
+++ /dev/null
@@ -1,47 +0,0 @@
-/* SPDX-License-Identifier:     GPL-2.0+
- *
- * Copyright (C) 2021 HabanaLabs Ltd.
- */
-
-#ifndef ARCH_GRECO_COMMON_GRECO_H_
-#define ARCH_GRECO_COMMON_GRECO_H_
-
-#include <goya2/goya2.h>
-
-#define ARC_RUN_ACK_TIMEOUT_US		100
-#define PSOC_AXI_DRAIN_TIMEOUT				0x4000 /* 16K cycles */
-#define PCIE_AXI_DRAIN_TIMEOUT				PSOC_AXI_DRAIN_TIMEOUT
-
-#define EFUSE_SYSCFG_WA_SNPS_CASE_01038891		BIT(23)
-
-#define PCIE_PHY_EXT_MPLLB_SSC_PEAK			0x5000
-#define PCIE_PHY_EXT_MPLLB_SSC_STEP_SIZE		0x19CE
-#define PCIE_PHY_EXT_MPLLA_SSC_PEAK			0x6666
-#define PCIE_PHY_EXT_MPLLA_SSC_STEP_SIZE		0x2108
-
-#define PCIE_PHY_PROTOCOL0_EXT_MPLLA_MULTIPLIER		0x50
-#define PCIE_PHY_PROTOCOL0_EXT_MPLLA_MULTIPLIER_SR	0xA0
-#define PCIE_PHY_PROTOCOL0_EXT_MPLLB_MULTIPLIER		0x7D
-#define PCIE_PHY_PROTOCOL0_EXT_TX_EQ_MAIN_G1_SLICE0	0x30
-#define PCIE_PHY_PROTOCOL0_EXT_TX_EQ_MAIN_G2_SLICE0	0x30
-#define PCIE_PHY_PROTOCOL0_EXT_TX_EQ_MAIN_G3_SLICE0	0x30
-#define PCIE_PHY_PROTOCOL0_EXT_TX_EQ_MAIN_G4_SLICE0	0x30
-
-#define PCIE_PHY_PROTOCOL0_EXT_MPLLA_BANDWIDTH_SR_LOW	0x25
-#define PCIE_PHY_PROTOCOL0_EXT_MPLLA_BANDWIDTH_SR_HIGH	0x25
-#define PCIE_PHY_PROTOCOL0_EXT_MPLLB_BANDWIDTH_SR_LOW	0x25
-#define PCIE_PHY_PROTOCOL0_EXT_MPLLB_BANDWIDTH_SR_HIGH	0x25
-#define PCIE_PHY_PROTOCOL0_EXT_MPLLB_BANDWIDTH_LOW	0x1C3
-#define PCIE_PHY_PROTOCOL0_EXT_MPLLB_BANDWIDTH_HIGH	0x1C3
-
-#define PCIE_PHY_PROTOCOL0_EXT_RX_ADAPT_MODE_G3_CONF	0x5555
-#define PCIE_PHY_PROTOCOL0_EXT_RX_ADAPT_SEL_G4_CONF	0xFF
-
-#define ARC0_MEM_SIZE			(ARC0_ICCM_SIZE + ARC0_DCCM_SIZE)
-#define ARC1_MEM_SIZE			(ARC1_ICCM_SIZE + ARC1_DCCM_SIZE)
-
-void run_arc(u8 id);
-void halt_n_reset_arc(u8 id);
-void axi_enable_drain(void);
-
-#endif /* ARCH_GRECO_COMMON_GRECO_H_ */
diff --git a/arch/greco/common/hl.h b/arch/greco/common/hl.h
new file mode 100644
index 0000000..21ed3d8
--- /dev/null
+++ b/arch/greco/common/hl.h
@@ -0,0 +1,47 @@
+/* SPDX-License-Identifier:     GPL-2.0+
+ *
+ * Copyright (C) 2021 HabanaLabs Ltd.
+ */
+
+#ifndef ARCH_GRECO_COMMON_HL_H_
+#define ARCH_GRECO_COMMON_HL_H_
+
+#include <goya2/goya2.h>
+
+#define ARC_RUN_ACK_TIMEOUT_US		100
+#define PSOC_AXI_DRAIN_TIMEOUT				0x4000 /* 16K cycles */
+#define PCIE_AXI_DRAIN_TIMEOUT				PSOC_AXI_DRAIN_TIMEOUT
+
+#define EFUSE_SYSCFG_WA_SNPS_CASE_01038891		BIT(23)
+
+#define PCIE_PHY_EXT_MPLLB_SSC_PEAK			0x5000
+#define PCIE_PHY_EXT_MPLLB_SSC_STEP_SIZE		0x19CE
+#define PCIE_PHY_EXT_MPLLA_SSC_PEAK			0x6666
+#define PCIE_PHY_EXT_MPLLA_SSC_STEP_SIZE		0x2108
+
+#define PCIE_PHY_PROTOCOL0_EXT_MPLLA_MULTIPLIER		0x50
+#define PCIE_PHY_PROTOCOL0_EXT_MPLLA_MULTIPLIER_SR	0xA0
+#define PCIE_PHY_PROTOCOL0_EXT_MPLLB_MULTIPLIER		0x7D
+#define PCIE_PHY_PROTOCOL0_EXT_TX_EQ_MAIN_G1_SLICE0	0x30
+#define PCIE_PHY_PROTOCOL0_EXT_TX_EQ_MAIN_G2_SLICE0	0x30
+#define PCIE_PHY_PROTOCOL0_EXT_TX_EQ_MAIN_G3_SLICE0	0x30
+#define PCIE_PHY_PROTOCOL0_EXT_TX_EQ_MAIN_G4_SLICE0	0x30
+
+#define PCIE_PHY_PROTOCOL0_EXT_MPLLA_BANDWIDTH_SR_LOW	0x25
+#define PCIE_PHY_PROTOCOL0_EXT_MPLLA_BANDWIDTH_SR_HIGH	0x25
+#define PCIE_PHY_PROTOCOL0_EXT_MPLLB_BANDWIDTH_SR_LOW	0x25
+#define PCIE_PHY_PROTOCOL0_EXT_MPLLB_BANDWIDTH_SR_HIGH	0x25
+#define PCIE_PHY_PROTOCOL0_EXT_MPLLB_BANDWIDTH_LOW	0x1C3
+#define PCIE_PHY_PROTOCOL0_EXT_MPLLB_BANDWIDTH_HIGH	0x1C3
+
+#define PCIE_PHY_PROTOCOL0_EXT_RX_ADAPT_MODE_G3_CONF	0x5555
+#define PCIE_PHY_PROTOCOL0_EXT_RX_ADAPT_SEL_G4_CONF	0xFF
+
+#define ARC0_MEM_SIZE			(ARC0_ICCM_SIZE + ARC0_DCCM_SIZE)
+#define ARC1_MEM_SIZE			(ARC1_ICCM_SIZE + ARC1_DCCM_SIZE)
+
+void run_arc(u8 id);
+void halt_n_reset_arc(u8 id);
+void axi_enable_drain(void);
+
+#endif /* ARCH_GRECO_COMMON_HL_H_ */
